user-guide/modules/ROOT/pages/advanced-builds.adoc

```diff
@@ -1,3 +1,247 @@
+
+In the context of pass:[C++], this generally comes up with GCC 5.x and later versions, which introduced a new ABI to conform more closely with the pass:[C++11] and pass:[C++14] standards. This new ABI changed the implementations of `std::string` and `std::list`, among other things, which broke binary compatibility with older versions of these libraries. This means that code compiled with the new ABI may _not_ be binary-compatible with code compiled with the old ABI.
+
+To build Boost with the old ABI when using GCC, you'd pass in the `-D_GLIBCXX_USE_CXX11_ABI=0` flag. For example:
+
+[source,txt]
+----
+./b2 toolset=gcc cxxflags="-D_GLIBCXX_USE_CXX11_ABI=0"
+----
+
+This command specifies the compiled Boost libraries are to be linked with code that was also compiled with the old ABI. Setting the flag to 1 will specify the new ABI, though of course this is the default and the flag is unnecessary.
+
+Check compatibility with your entire toolchain and codebase when making decisions about which ABI to use. It's generally best to use the new ABI if all your code and dependencies support it.
+
+== Specify the Standard Library Implementation
+
+When using Boost libraries in a project, you might choose to use `pass:[libc++]` or `pass:[libstdc++]`. These are two different standard library implementations that you can use with your pass:[C++] compiler. The choice between the two depends on the compiler you are using, the platforms you are targeting, the standards and features you are using, and the specific requirements of your project.
+
+. `pass:[libc++]` is the standard library implementation developed by the https://github.com/llvm/llvm-project[LLVM project]. It is the default library for the Clang compiler and is known for its performance and standards compliance. It's also designed to work well with the Clang static analyzer and other LLVM tools. One key feature of `pass:[libc++]` is that it supports deployment to older versions of macOS and iOS.
+
+. `pass:[libstdc++]` is the GNU Standard Library, which comes as the default with the GCC compiler. It is a robust and mature library, supporting many pass:[C++] standards and extensions. It's used by a large number of projects and is available on a wide variety of systems.
+
+You can specify which standard library you want to use by setting the appropriate flags in your build system. This is dependent on the compiler you're using.
+
+If you're using the Clang compiler and you want to use `pass:[libc++]`, you can use the `-stdlib=libc++` compiler flag. For example:
+
+[source,txt]
+----
+clang++ -stdlib=libc++ YOUR_FILE.cpp -lboost_system
+----
+
+If you're using GCC and you want to use  `pass:[libstdc++]`, you don't need to do anything specific because this is the default standard library for GCC. If you're using Clang and want to use  `pass:[libstdc++]`, you can use the `-stdlib=libstdc++` flag:
+
+[source,txt]
+----
+clang++ -stdlib=libstdc++ your_file.cpp -lboost_system
```

Comment:
The text mentions GCC and the example uses clang.

I'm not even sure the GCC flags are the same. 

I've only ever used `-stdlib` with clang to choose between libc++ and libstdc++ because clang provides libc++ but it defaults to libstdc++ on linux to avoid incompatibilities with GCC. I have never messed with that for any other reason and always use the default std implementation otherwise.